### PR TITLE
DPI card + sheet

### PR DESCRIPTION
**Pull request by brunobar79**
PoW: https://recordit.co/mFkfkcEq86

### CODE REVIEW

File: src/components/coin-row/UnderlyingAssetCoinRow.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  address,
+  change,
+  isPositive,
+  name,
+  onPress,
+  price,
+  symbol,
+}) => {
+  const handlePress = useCallback(() => {
+    onPress?.({ address, change, name, price, symbol });
```
Comment (osdnk) [inline src/components/coin-row/UnderlyingAssetCoinRow.js:21]:
```suggestion
    onPress({ address, change, name, price, symbol });
```
I think that everywhere `onPress` present

File: src/components/discover-sheet/TopMoversSection.js [hunk 2]
```diff
@@ -66,10 +66,10 @@ export default function TopMoversSection() {
 
       <Column>
         {gainerItems?.length !== 0 && (
-          <MarqueeList items={gainerItems} speed={300} />
+          <MarqueeList items={gainerItems} speed={666} />
```
Comment (osdnk) [inline src/components/discover-sheet/TopMoversSection.js:69]: 😈

File: src/components/expanded-state/TokenIndexExpandedState.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  const { height: screenHeight } = useDimensions();
+
+  return (
+    <Fragment>
+      <SlackSheet
+        bottomInset={42}
+        {...(ios
+          ? { height: '100%' }
+          : { additionalTopPadding: true, contentHeight: screenHeight - 80 })}
+        scrollEnabled
```
Comment (osdnk) [inline src/components/expanded-state/TokenIndexExpandedState.js:153]: isn't this default?

File: src/components/qrcode-scanner/QRCodeScanner.js [hunk 4]
```diff
@@ -60,7 +60,7 @@ export default function QRCodeScanner({
     [setCameraEnabled]
   );
   useEffect(
-    () => dsRef.current.addOnCrossMagicBorderListener(onCrossMagicBorder),
+    () => dsRef.current?.addOnCrossMagicBorderListener(onCrossMagicBorder),
```
Comment (osdnk) [inline src/components/qrcode-scanner/QRCodeScanner.js:63]: it doesn't work on Android?

File: src/references/defisdk/index.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export { default as defiSdkAdapterRegistryABI } from './adapterRegistryABI.json';
+
+export const DEFI_SDK_ADAPTER_REGISTRY_ADDRESS =
```
Comment (osdnk) [inline src/references/defisdk/index.ts:3]: can we move this to some env?

File: src/components/expanded-state/TokenIndexExpandedState.js (hunk 3 above)
Comment (brunobar79) [reply src/components/expanded-state/TokenIndexExpandedState.js:172]: yep. removing...

File: src/components/qrcode-scanner/QRCodeScanner.js (hunk 4 above)
Comment (brunobar79) [reply src/components/qrcode-scanner/QRCodeScanner.js:63]: Christian found a crash while having discover disabled / or enabled (don't remember) and that prevent it

File: src/references/defisdk/index.ts (hunk 5 above)
Comment (brunobar79) [reply src/references/defisdk/index.ts:3]: I defer this one to @jinchung who wrote it originally.

File: src/components/qrcode-scanner/QRCodeScanner.js (hunk 4 above)
Comment (osdnk) [reply src/components/qrcode-scanner/QRCodeScanner.js:63]: It is interesting. Can we check why?

**Comment by osdnk** [comment]
@brunobar79 here are some conflicts that are probably made by my merge commit and I am sorry.

**Review by skylarbarrera — APPROVED** [review]
LGTM, a few things i noticed that may be out of scope, in order of importance/scope imo.

We hide DPI if ANY wallet that is imported has DPI, not just the current wallet

We don't check if underlying assets are in the wallet for their expanded state

Sheet height could be a little taller (last asset is nearly cut off): http://cloud.skylarbarrera.com/Screen-Shot-2021-02-02-14-05-01.98.png

Sheet Swiping doesnt work sometimes, works 7/10 I'd say

We should show DPI sheet over normal Expanded State on wallet screen

Invisible Chart Bug seem's to be more prevalent here 😞

**Comment by brunobar79** [comment]
@skylarbarrera  Pushed the fix for all the things you mentioned and I'm gonna follow up with the scroll issues with Michal later.  Merging this and we can follow up on the discover branch.

**Pull request merged**